srml/grandpa/src/lib.rs

```diff
@@ -242,8 +251,13 @@ decl_module! {
 
 impl<T: Trait> Module<T> {
 	/// Get the current set of authorities, along with their respective weights.
-	pub fn grandpa_authorities() -> Vec<(AuthorityId, AuthorityWeight)> {
-		Authorities::get()
+	pub fn grandpa_authorities() -> AuthorityList {
+		storage::unhashed::get_or_default(GRANDPA_AUTHORITIES_KEY)
```

Comment:
I'm possibly wrong, but let's say we have upgraded runtime on block B. Until next block (i.e. until `on_initialize` will be called), the value at well known key will be empty. Is it true?

Reply:
Yes, I think that's right.